File: projetmanga-server/models/Store.js
const mongoose = require('mongoose');

const StoreSchema = new mongoose.Schema({
    nom: { type: String, required: true, trim: true },
    adresse: String,
    telephone: String,
    email: String,
    horaires: String,

    position: {
        type: { type: String, enum: ['Point'], default: 'Point' },
        coordinates: { type: [Number], required: true }
    }

}, { timestamps: true });

StoreSchema.index({ position: "2dsphere" });

module.exports = mongoose.model('Store', StoreSchema);

File: projetmanga-client-android/app/src/main/java/com/mangaproject/data/model/Store.kt
package com.mangaproject.data.model

import com.google.gson.annotations.SerializedName

data class Store(
    @SerializedName("_id")
    val id: String,
    val nom: String,
    val adresse: String?,
    val telephone: String?,
    val email: String?,
    val position: Position
)

data class Position(
    val type: String,
    val coordinates: List<Double>  // [lon, lat]
)
